### ISSUE TITLE
to_netcdf fails for engine=h5netcdf when using dask-backed arrays

### ISSUE THREAD

**Issue by jcmgray** [pydata/xarray#1571]
When using dask-backed datasets/arrays it does not seem possible to use the 'h5netcdf' engine to write to disk:

```python
import xarray as xr
ds = xr.Dataset({'a': ('x', [1, 2])}, {'x': [3, 4]}).chunk()
ds.to_netcdf("test.h5", engine='h5netcdf')
```

results in the error:

```bash
...

h5py/h5a.pyx in h5py.h5a.open()

KeyError: "Can't open attribute (can't locate attribute: 'dask')"
```

Not sure if this is a xarray or h5netcdf issue - or some inherent limitation in which case apologies!

**Comment by shoyer** [comment]
Thanks for the report!

This was definitely working at one point, but it appears we did not have test coverage and then it was broken by https://github.com/dask/dask/pull/2181.

I'm going to fix this on the h5netcdf side, but will leave this issue open as a reminder to write an integration test to catch issues like this.

**Comment by shoyer** [comment]
This should be fixed in the latest h5netcdf release (v0.4.2), which is now up on pypi (so you can install it with `pip`).
